### PR TITLE
NIFI-7580-Add documentation around autoloading NARs

### PR DESCRIPTION
**Pull request by sdairs**
Adds documenation to the Administration guide for installing custom processors & using the autoload features
https://issues.apache.org/jira/browse/NIFI-7580

### CODE REVIEW

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 1]
```diff
@@ -3676,3 +3676,65 @@ In your new NiFi installation:
...
+[[processor-location-options]]
+=== Available Configuration Options
+
+NiFi provides 3 configuration options for processor locations. Namely:
+
+   nifi.nar.library.directory
+   nifi.nar.library.directory.<custom>
+   nifi.nar.library.autoload.directory
+
+NOTE: Paths set using these options are relative to the NiFi Home Directory. For example, if the NiFi Home Directory is `/var/lib/nifi`, and the Library Directory is `/lib`, then the final path is `/var/lib/nifi/lib`.
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3692]:
```suggestion
NOTE: Paths set using these options are relative to the NiFi Home Directory. For example, if the NiFi Home Directory is `/var/lib/nifi`, and the Library Directory is `./lib`, then the final path is `/var/lib/nifi/lib`.
```

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 2]
```diff
@@ -3676,3 +3676,65 @@ In your new NiFi installation:
...
+
+The `nifi.nar.library.directory` is used for the default location for provided NiFi processors. It is not recommended to use this for custom processors as these could be lost during a NiFi upgrade.
+For example:
+   nifi.nar.library.directory=/lib
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3696]:
```suggestion
The `nifi.nar.library.directory` is used for the default location for provided NiFi processors. It is not recommended to use this for custom processors as these could be lost during a NiFi upgrade. For example:

   nifi.nar.library.directory=/lib
```

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 3]
```diff
@@ -3676,3 +3676,65 @@ In your new NiFi installation:
...
+
+The `nifi.nar.library.directory.<custom>` allows the admin to provide multiple arbritary paths for NiFi to locate custom processors. A unique property identifier must append the property for each unique path.
+For example:
+   nifi.nar.library.directory.myCustomLibs=/my-custom-nars/lib
+   nifi.nar.library.directory.otherCustomLibs=/other-custom-nars/lib
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3701]:
```suggestion
The `nifi.nar.library.directory.<custom>` allows the admin to provide multiple arbritary paths for NiFi to locate custom processors. A unique property identifier must append the property for each unique path. For example:

   nifi.nar.library.directory.myCustomLibs=./my-custom-nars/lib
   nifi.nar.library.directory.otherCustomLibs=./other-custom-nars/lib
```

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 4]
```diff
@@ -3676,3 +3676,65 @@ In your new NiFi installation:
...
+
+The `nifi.nar.library.autoload.directory` is used by the autoload feature, where NiFi can automatically load new processors added to the configured path without requiring a restart.
+For example:
+   nifi.nar.library.autoload.directory=/autoload/lib
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3705]:
```suggestion
The `nifi.nar.library.autoload.directory` is used by the autoload feature, where NiFi can automatically load new processors added to the configured path without requiring a restart. For example:

   nifi.nar.library.autoload.directory=./autoload/lib
```

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 5]
```diff
@@ -3676,3 +3676,65 @@ In your new NiFi installation:
...
+
+=== Installing Custom Processors
+
+This section describes the original process for installing custom processors that requires a restart to NiFi. To use the Autoloading feature, see the below <<autoloading-processors>> section.
+
+Firstly, we will configure a directory for the custom processors. See <<processor-location-options>> for more about these configuration options.
+
+   nifi.nar.library.directory.myCustomLibs=/my-custom-nars/lib
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3713]:
```suggestion
   nifi.nar.library.directory.myCustomLibs=./my-custom-nars/lib
```

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 6]
```diff
@@ -3676,3 +3676,65 @@ In your new NiFi installation:
...
+
+Restart NiFi and the custom processor should now be available when adding a new Processor to your flow.
+
+[[autoloading-processors]]
+=== Autoloading Custom Processors
+
+This section describes the process to use the Autoloading feature for custom processors.
+
+To use the autoloading feature, the `nifi.nar.library.autoload.directory` property must be configured to point at the desired directory.
+By default, this points at `./extensions`.
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3729]:
```suggestion
To use the autoloading feature, the `nifi.nar.library.autoload.directory` property must be configured to point at the desired directory. By default, this points at `./extensions`. 
```

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 7]
```diff
@@ -3676,3 +3676,65 @@ In your new NiFi installation:
...
+[[autoloading-processors]]
+=== Autoloading Custom Processors
+
+This section describes the process to use the Autoloading feature for custom processors.
+
+To use the autoloading feature, the `nifi.nar.library.autoload.directory` property must be configured to point at the desired directory.
+By default, this points at `./extensions`. 
+
+For example:
+   nifi.nar.library.autoload.directory=./extensions
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3732]:
```suggestion
For example:

   nifi.nar.library.autoload.directory=./extensions
```

**Review by pvillard31 — CHANGES_REQUESTED** [review]
Left a few comments to make sure the rendering is correct. Otherwise we would have:

Screenshot 2020-09-15 at 14 33 09

**Comment by pvillard31** [comment]
From a content point of view, it LGTM but since the JIRA was assigned to @alopresto, requesting his review to make sure we cover everything or if follow up JIRAs are required.

**Comment by alopresto** [comment]
Unfortunately with some competing priorities, I don't have time to look at this now. Assuming the docs don't make things _more_ confusing, something is better than nothing in my mind. Thanks for contributing this. Pierre, please proceed without me.

File: nifi-docs/src/main/asciidoc/administration-guide.adoc [hunk 8]
```diff
@@ -3676,3 +3676,65 @@ In your new NiFi installation:
...
+
+   nifi.nar.library.directory
+   nifi.nar.library.directory.<custom>
+   nifi.nar.library.autoload.directory
+
+NOTE: Paths set using these options are relative to the NiFi Home Directory. For example, if the NiFi Home Directory is `/var/lib/nifi`, and the Library Directory is `./lib`, then the final path is `/var/lib/nifi/lib`.
+
+The `nifi.nar.library.directory` is used for the default location for provided NiFi processors. It is not recommended to use this for custom processors as these could be lost during a NiFi upgrade. For example:
+
+   nifi.nar.library.directory=/lib
```
Comment (pvillard31) [inline nifi-docs/src/main/asciidoc/administration-guide.adoc:3696]:
```suggestion
   nifi.nar.library.directory=./lib
```

**Review by pvillard31 — APPROVED** [review]

**Comment by pvillard31** [comment]
Merged to main, not closing the JIRA as the versioning consideration has not been included and might be important in the context of a development cluster where someone would be trying changes in a custom NAR without changing the name of it (and this would not be auto-loaded as a new version of the NAR).

**Pull request closed without merging**